app/models/service_template_container_template.rb

```diff
@@ -0,0 +1,90 @@
+class ServiceTemplateContainerTemplate < ServiceTemplateGeneric
+  def self.default_provisioning_entry_point(_service_type)
+    '/Service/Generic/StateMachines/GenericLifecycle/provision'
+  end
+
+  def self.default_retirement_entry_point
+    nil
+  end
+
+  # create ServiceTemplate and supporting ServiceResources and ResourceActions
+  # options
+  #   :name
+  #   :description
+  #   :service_template_catalog_id
+  #   :config_info
+  #     :provision
+  #       :dialog_id or :dialog
+  #       :container_template_id or :container_template
+  #
+  def self.create_catalog_item(options)
+    options     = options.merge(:service_type => 'atomic', :prov_type => 'generic_container_template')
+    config_info = validate_config_info(options[:config_info])
+    enhanced_config = config_info.deep_merge(
+      :provision => {
+        :configuration_template => container_template_from_config_info(config_info)
+      }
+    )
+
+    transaction do
+      create_from_options(options).tap do |service_template|
+        service_template.create_resource_actions(enhanced_config)
+      end
+    end
+  end
+
+  def self.validate_config_info(info)
+    info[:provision][:fqname] ||= default_provisioning_entry_point('atomic') if info.key?(:provision)
```

Comment:
For `Ansible PlayBook` services we are not exposing the provisioning entry point so this check would make sense there, but here the UI will allow for different automate entry points to be selected.  cc @bzwei